website/blog/2021-02-22-jest-27.md

```diff
@@ -0,0 +1,78 @@
+
+<!-- TODO pending merge https://github.com/facebook/jest/pull/9351 -->
+We've also merged [a PR](https://github.com/facebook/jest/pull/9351) to be able to deal with test files symlinked into the test directory, a feature much wanted by users of [Bazel](https://bazel.build/).
+
+<!-- TODO pending merge https://github.com/facebook/jest/issues/9504 -->
+[Another PR](https://github.com/facebook/jest/issues/9504) enabled [`transform`s](/docs/en/configuration#transform-objectstring-pathtotransformer--pathtotransformer-object) to be asynchronous, a requirement to support transpilation through tools such as [esbuild](https://esbuild.github.io/), [Snowpack](https://www.snowpack.dev/), and [Vite](https://www.snowpack.dev/) effectively.
+
+## Flipping defaults
+
+Up until now, if you were using Jest in its default configuration, you were—perhaps unknowingly—running some code forked many years ago from the test runner [Jasmine 2.0](https://jasmine.github.io/2.0/introduction) that provides test framework functions such as `describe`, `it`, and `beforeEach`. In 2017, [Aaron Abramov](https://github.com/aaronabramov) [initially wrote](https://github.com/facebook/jest/pull/3668) a **replacement for the Jasmine code** called `jest-circus`, meant to improve error messages, maintainability, and extensibility.  
+After years of large-scale use at Facebook and of course in Jest itself, as well as recent adoption in `create-react-app`, we are now confident that `jest-circus` is highly compatible with `jest-jasmine2` and should work in most environments with little to no migration work. There may be minor differences in execution order and strictness, but we expect no major upgrade difficulties other than for code relying on Jasmine-specific APIs such as `jasmine.getEnv()`. If you rely extensively on such APIs, you can opt back in to the Jasmine-based test runner by [configuring](/docs/en/configuration#testrunner-string) `"testRunner": "jest-jasmine2"`.
+
+Running tests in a [JSDOM environment](/docs/en/configuration#testenvironment-string) incurs a significant performance overhead. Because this was the default behavior of Jest unless otherwise configured up until now, users who are writing Node apps, for example, may not even know they are given an expensive DOM environment that they do not even need.  
+For this reason, we are **changing the default test environment** from `"jsdom"` to `"node"`. If you are affected by this change because you use DOM APIs and do not have the test environment explicitly configured, you should be receiving an error when e.g. `document` is accessed, and you can configure `"testEnvironment": "jsdom"` or use [per-file environment configuration](/docs/en/configuration#testenvironment-string) to resolve this.  
```

Comment:
```suggestion
For this reason, we are **changing the default test environment** from `"jsdom"` to `"node"`. If you are affected by this change because you use DOM APIs and do not have the test environment explicitly configured, you should be receiving an error when e.g. `document` is accessed, and you can configure `"testEnvironment": "jsdom"` or use [per-file environment configuration](Configuration#testenvironment-string) to resolve this.  
```

is the generated html linking correctly with this?

this might also be different with docusaurs 2, I'll try to land that PR this coming week

Reply:
Yeah let's wait for docu v2, gotta double check how it looks once that is done anyway

Reply:
Nope, doesn't

Reply:
It should work if you include `.md`

Reply:
And if not, should remove the `en` to skip a redirect

Reply:
There's no `en` on any link other than the one to `nodejs.org` o.O that was pre Docu v2

Reply:
And nope, that just gets you http://localhost:3000/blog/2021/03/11/Configuration.md#testenvironment-string

Reply:
You get cannot resolve errors in the console. Could be because blog isn't the same folder idk

Reply:
ah, might be 👍